packages/logging/src/v2/logging_service_v2_api.js

```diff
@@ -0,0 +1,454 @@
+  }
+  if ('partialSuccess' in args.otherArgs) {
+    req.partial_success = args.otherArgs.partialSuccess;
+  }
+  return this._writeLogEntries(req, args.options, args.callback);
+};
+
+/**
+ * Lists log entries.  Use this method to retrieve log entries from Cloud
+ * Logging.  For ways to export log entries, see
+ * [Exporting Logs](https://cloud.google.com/logging/docs/export).
+ *
+ * @param {string[]} projectIds
+ *   Required. One or more project IDs or project numbers from which to retrieve
+ *   log entries.  Examples of a project ID: `"my-project-1A"`, `"1234567890"`.
+ * @param {Object=} otherArgs
+ * @param {string=} otherArgs.filter
+ *   Optional. An [advanced logs filter](https://cloud.google.com/logging/docs/view/advanced_filters).
+ *   The filter is compared against all log entries in the projects specified by
+ *   `projectIds`.  Only entries that match the filter are retrieved.  An empty
+ *   filter matches all log entries.
+ * @param {string=} otherArgs.orderBy
+ *   Optional. How the results should be sorted.  Presently, the only permitted
+ *   values are `"timestamp asc"` (default) and `"timestamp desc"`. The first
+ *   option returns entries in order of increasing values of
+ *   `LogEntry.timestamp` (oldest first), and the second option returns entries
+ *   in order of decreasing timestamps (newest first).  Entries with equal
+ *   timestamps are returned in order of `LogEntry.insertId`.
+ * @param {number=} otherArgs.pageSize
+ *   The maximum number of resources contained in the underlying API
+ *   response. If page streaming is performed per-resource, this
+ *   parameter does not affect the return value. If page streaming is
+ *   performed per-page, this determines the maximum number of
+ *   resources in a page.
+ * @param {gax.CallOptions=} options
+ *   Overrides the default settings for this call, e.g, timeout,
+ *   retries, etc.
+ * @returns {Stream}
+ *   An object stream. By default, this emits an object representing
+ *   [LogEntry]{@link LogEntry} on 'data' event.
+ *   This object can also be configured to emit
+ *   pages of the responses through the options parameter.
+ *
+ * @example
+ *
+ * var api = loggingV2.loggingServiceV2Api();
+ * var projectIds = [];
+ * api.listLogEntries(projectIds).on('data', function(element) {
+ *     // doThingsWith(element)
+ * });
+ */
+LoggingServiceV2Api.prototype.listLogEntries = function listLogEntries() {
+  var args = arguejs({
+    projectIds: Array,
+    otherArgs: [Object, {}],
+    options: [gax.CallOptions],
+    callback: [Function]
+  }, arguments);
+  var req = {
+    project_ids: args.projectIds
+  };
+  if ('filter' in args.otherArgs) {
+    req.filter = args.otherArgs.filter;
+  }
+  if ('orderBy' in args.otherArgs) {
+    req.order_by = args.otherArgs.orderBy;
+  }
+  if ('pageSize' in args.otherArgs) {
+    req.page_size = args.otherArgs.pageSize;
+  }
+  return this._listLogEntries(req, args.options, args.callback);
```

Comment:
Yes, it returns a stream and it automatically does the automatic pagination (it's called page-streaming, in our side). It automatically fetches the next response by using `nextPageToken` and emits an instance of log entry instead of listLogEntriesResponse upon 'data' event.

In order to iterate over pages, it actually require to specify a value in the call options.
```javascript
var v2 = require('@google-cloud/logging').v2();
var gax = require('google-gax');
var api = v2.loggingServiceV2Api();

api.listLogEntries([projectId], new gax.CallOptions({page_token: gax.FIRST_PAGE})).on('data', function(resp) {
   // resp is a listLogEntriesResponse.
}).on('end', function() { ...});
```
`page_token` here will be supplied from the last result, or use `gax.FIRST_PAGE` object when starting from a page.

This design borrows from what Python is doing, so might not look like the way JS programmers will do. In that case please tell me how this can be improved.